exercises/palindrome-products/src/lib.rs

```diff
@@ -0,0 +1,17 @@
+#[derive(Debug, PartialEq)]
+pub enum Error{
+    Empty,
+    RangeFailure,
+}
+#[derive(Debug, PartialEq)]
+pub struct ReturnVals{
+    pub result: i32,
```

Comment:
No negative number can be a palindrome, so let's use an unsigned type for the palindromes.

Reply:
Let's use `u64`, in fact, because it's easier than you might expect to overflow a `u32`. It's fine if `min` and `max` are `u32`, however; this helps reduce the chance of an overflow.